pytest_girder/pytest_girder/fixtures.py

```diff
@@ -66,6 +79,11 @@ def db(request):
 
     connection.close()
 
+    # Clear connection cache and model singletons
+    _dbClients.clear()
+    for model in model_base._modelSingletons:
+        model.__class__._instance = None
```

Comment:
@zachmullen I hadn't realized that the implications of this before it was merged.

If a singleton binds events, those events will still be bound after the singleton is reset.  This makes it so tests involving such singletons will have to unbind events so that they can be rebound to the new singleton.

If we reset the singletons, should we also somehow unbind the events on that singleton?  Or add a convenience fixture for unbinding events with a particular handler name?

Reply:
Event binding is now idempotent, so doing the same binding multiple times should be fine. Did you run into a case where this actually breaks?

Reply:
Yes -- in large_image_annotation tests.  I bind to an event to watch a settings change.  I see this event triggered on an instance of the model, but another instance of the model accessed from the test.  Since we don't allow rebinding to an event with the same same eventName and handlerName combination, I see warnings about that.  Unbinding all of the large_image specific events (by, for instance, walking through the events and unbinding anything that matches the handlerName) between each test avoids the problem.

Reply:
> Since we don't allow rebinding to an event with the same same eventName and handlerName combination

Ah, I assumed it would override the existing binding, which may be more desirable behavior. Barring that, I can't think of a robust solution for the test fixture that would automagically unbind all events that were bound during model instantiation.

Reply:
To work around this, I'm unbinding all handlers with a specific handlerName.  Unfortunately, this requires reaching in to `events._mapping`, which, being a private member, is clearly not intended.  It seems like we should either make doing that an official method (at least in the pytest utilities) or have rebinding an event replace the existing binding (it could still produce a warning).

Reply:
I'm in favor of having later bindings replace earlier ones and emitting a warning, that seems a better and more expected behavior than ignoring the later binding.